driver/pelican/types/thermSchedule.go

```diff
@@ -0,0 +1,238 @@
+		}
+
+		schedules[thermostatID.Id] = thermSchedule
+	}
+	return schedules, nil
+}
+
+func (pel *Pelican) getSettings(sitename, thermostatID string, cookie *http.Cookie) (*settingsWrapper, error) {
+	var requestURL bytes.Buffer
+	requestURL.WriteString(fmt.Sprintf("https://%s.officeclimatecontrol.net/ajaxThermostat.cgi?id=", sitename))
+	requestURL.WriteString(thermostatID)
+	requestURL.WriteString(":Thermostat&request=GetSchedule")
+
+	resp, _, errs := pel.scheduleReq.Get(requestURL.String()).Type("form").AddCookie(cookie).End()
+	if errs != nil {
+		return nil, fmt.Errorf("Failed to retrieve schedule settings for thermostat %v: %v", thermostatID, errs)
+	}
+	var result settingsRequest
+	decoder := json.NewDecoder(resp.Body)
+	if decodeError := decoder.Decode(&result); decodeError != nil {
+		return nil, fmt.Errorf("Failed to decode schedule settings for thermostat %v: %v", thermostatID, decodeError)
+	}
+	return &result.Userdata, nil
+}
+
+func (pel *Pelican) getScheduleByDay(dayOfWeek int, epnum float64, sitename, thermostatID string, cookie *http.Cookie, timezone *time.Location) (*[]ThermostatBlockSchedule, error) {
+	// Construct Request URL for Thermostat Schedule by Day of Week
+	var requestURL bytes.Buffer
+	requestURL.WriteString(fmt.Sprintf("https://%s.officeclimatecontrol.net/thermDayEdit.cgi?section=json&nodename=", sitename))
+	requestURL.WriteString(thermostatID)
+	requestURL.WriteString("&epnum=")
+	requestURL.WriteString(fmt.Sprintf("%.0f", epnum))
+	requestURL.WriteString("&dayofweek=")
+	requestURL.WriteString(strconv.Itoa(dayOfWeek))
+
+	// Make Request, Decode into Response Struct
+	resp, _, errs := pel.scheduleReq.Get(requestURL.String()).Type("form").AddCookie(cookie).End()
+	if errs != nil {
+		return nil, fmt.Errorf("Failed to retrieve schedule for thermostat %v on day of week %v: %v", thermostatID, dayOfWeek, errs)
+	}
+	var result scheduleRequest
+	decoder := json.NewDecoder(resp.Body)
+	if decodeError := decoder.Decode(&result); decodeError != nil {
+		return nil, fmt.Errorf("Failed to decode schedule for thermostat %v on day of week %v: %v", thermostatID, dayOfWeek, decodeError)
+	}
+
+	// Transfer Response Struct Data into return struct
+	var daySchedule []ThermostatBlockSchedule
+	for _, block := range result.ClientData.SetTimes {
+		returnBlock := ThermostatBlockSchedule{
+			CoolSetting: block.CoolSetting,
+			HeatSetting: block.HeatSetting,
+			System:      block.System,
+		}
+
+		if rruleTime, rruleError := convertTimeToRRule(dayOfWeek, block.StartValue, timezone); rruleError != nil {
+			return nil, fmt.Errorf("Failed to convert time in string format %v to rrule format: %v", block.StartValue, rruleError)
+		} else {
+			returnBlock.Time = rruleTime
+		}
+
+		daySchedule = append(daySchedule, returnBlock)
+	}
+	return &daySchedule, nil
+}
+
+func convertTimeToRRule(dayOfWeek int, blockTime string, timezone *time.Location) (string, error) {
+	timeParsed, timeParsedErr := time.Parse("15:04:PM", blockTime)
```

Comment:
That's interesting, there's a trailing colon after the minutes?

Reply:
Yeah, weirdly enough the format with which the time is returned has that trailing colon. However, it seems to parse the time correctly (e.g. if I'm parsing 6:00:PM, for hour, "timeParsed" returns it as 18, which is what we want.